platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -2646,9 +2646,9 @@ static void tm_system_capabilities_set(odp_tm_capabilities_t *cap_ptr,
 		max_priority     = MIN(per_level_req->max_priority,
 				       ODP_TM_MAX_PRIORITIES - 1);
 		min_weight       = MAX(per_level_req->min_weight,
-				       ODP_TM_MIN_SCHED_WEIGHT);
+				       (uint32_t)ODP_TM_MIN_SCHED_WEIGHT);
 		max_weight       = MIN(per_level_req->max_weight,
-				       ODP_TM_MAX_SCHED_WEIGHT);
+				       (uint32_t)ODP_TM_MAX_SCHED_WEIGHT);
```

Comment:
These casts are a bit ugly. I wonder if ODP_TM_MAX_SCHED_WEIGHT should be defines as unsigned in the ABI instead, i.e.: #define ODP_TM_MAX_SCHED_WEIGHT  255U

It seems one issue here is that ODP API specifies various constants that the ABI headers should define but it does not not specify their type. It seems that most or all of such literals are signed currently.

Note also that there is a convention that linux-gen changes are not done in the same patch as API changes so that API patches stay free from code that is specific to any particular implementation.

Reply:
Ack, I'll change to 255U. After changing that, we atleast for now won't get signed/unsigned comparison warning.\

I'm also not sure how we can enforce types for constants